code/main/consent/consentModule.brs

```diff
@@ -17,99 +17,135 @@ function _adb_isConsentModule(module as object) as boolean
     return (module <> invalid and module.type = "com.adobe.module.consent")
 end function
 
-function _adb_ConsentModule(configurationModule as object) as object
-    if not _adb_isConfigurationModule(configurationModule) then
-        _adb_logError("ConsentModule::_adb_ConsentModule() - configurationModule is not valid.")
+function _adb_ConsentModule(consentState as object, edgeModule as object) as object
+    if not _adb_isConsentStateModule(consentState) then
+        _adb_logError("ConsentModule::_adb_ConsentModule() - consentState is not valid.")
+        return invalid
+    end if
+
+    if not _adb_isEdgeModule(edgeModule) then
+        _adb_logError("ConsentModule::_adb_ConsentModule() - edgeModule is not valid.")
         return invalid
     end if
 
     module = _adb_AdobeObject("com.adobe.module.consent")
     module.Append({
-        ''' TODO: Update the path
-        _CONSENT_REQUEST_PATH: "/ee/v1/interact",
-        _configurationModule: configurationModule,
-        _collectConsent: invalid,
-
-        ''' TODO
+        _CONSENT_REQUEST_PATH: "/v1/privacy/set-consent",
+        _HANDLE_TYPE_CONSENT_PREFERENCES: "consent:preferences",
+        _CONSENT_PENDING: "p",
+        _edgeModule: edgeModule,
+        _consentState: consentState,
+
+        ' setConsent API triggers this API to queue edge requests
+        ' event: event generated by the public API with consent data
         processEvent: function(event as object) as void
-            ''' Process the event and update the collect consent
-            ''' Update the collect consent in persistence
-            ''' Send the consent update request to Edge
-            ''' Update the collect consent in persistence based on the edge request response
-        end function,
+            _adb_logVerbose("ConsentModule::processEvent() - Received event:(" + chr(10) + FormatJson(event) + chr(10) + ")")
 
-        ''' TODO
-        processResponseEvent: function(event as object) as void
-            _adb_logInfo("ConsentModule::processResponseEvent() - Received response event:(" + chr(10) + FormatJson(event) + chr(10) + ")")
-            ''' Process the edge response event and update the collect consent from konductor response
-            ''' Update the collect consent in persistence
-        end function,
+            requestId = event.uuid
+            eventData = event.data
+            timestampInMillis = event.timestampInMillis
 
-        ''' TODO
-        setConsent: function(consent as object) as void
-            ''' Send the consent update request to Edge
-            ''' Update the collect consent in persistence based on the edge request response
-        end function,
+            collectConsentValue = m._extractCollectConsentValue(eventData)
+            if _adb_isEmptyOrInvalidString(collectConsentValue)
+                _adb_logVerbose("ConsentModule::processEvent() - collect consent value not found in data:(" + FormatJson(eventData) + ").")
+            end if
```

Comment:
I don't see a need for this check here as it doesn't change the flow of the function. If the intent is just to log the collect consent value, the log at line 59 will do that when the consent payload is logged before queuing.

Reply:
Yes, I have some updates on this logic will bundle with my next PR.

Reply:
How extensive are those changes? If you are making significant changes to the logic for `processEvent` it would make sense to include those changes in this PR.

Reply:
I have changes to to this plus other updates in Edge Module and other places to enforce the consent and allow sending consent requests. 

I have added you in slack channel regarding the workflow for allowing consent requests (I see consent update (collect, share etc..)  go out even when collect consent is set to n.